### PR TITLE
Update SCTP support to beta status

### PR DESCRIPTION
**Pull request by janosi**
SCTP support feature would move to "beta" stage. The documents should be updated to reflect this change.
The enhancement issue is tracked here: https://github.com/kubernetes/enhancements/issues/614
It requires new e2e test cases to move SCTP support to beta.

### CODE REVIEW

**Comment by sftim** [comment]
/hold
because (I presume) this feature promotion is waiting on the end-to-end tests merging: https://github.com/kubernetes/kubernetes/pull/88196

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -1139,9 +1139,9 @@ followed by the data from the client.
 
 ### SCTP
 
-{{< feature-state for_k8s_version="v1.12" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.18" state="beta" >}}
 
-Kubernetes supports SCTP as a `protocol` value in Service, Endpoint, NetworkPolicy and Pod definitions as an alpha feature. To enable this feature, the cluster administrator needs to enable the `SCTPSupport` feature gate on the apiserver, for example, `--feature-gates=SCTPSupport=true,…`.
+Kubernetes supports SCTP as a `protocol` value in Service, Endpoint, NetworkPolicy and Pod definitions as a beta feature. To enable this feature, the cluster administrator needs to enable the `SCTPSupport` feature gate on the apiserver, for example, `--feature-gates=SCTPSupport=true,…`.
```
[the remark this answers is not in the document]
Comment (janosi) [reply content/en/docs/concepts/services-networking/service.md]: Well, that is a good question. Once it moves to beta the feature flag is true by default, but it does not mean, that it is true in a cluster that I get from an administrator. So, it is still true from that perspective, that this flag must be true.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -124,7 +124,7 @@ different Kubernetes components.
 | `RunAsGroup` | `true` | Beta | 1.14 | |
 | `RuntimeClass` | `false` | Alpha | 1.12 | 1.13 |
 | `RuntimeClass` | `true` | Beta | 1.14 | |
-| `SCTPSupport` | `false` | Alpha | 1.12 | |
```
[the remark this answers is not in the document]
Comment (janosi) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Fixed.

File: content/en/docs/concepts/services-networking/service.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/services-networking/service.md]: Consider rewording, eg “as a beta feature, this is enabled by default. You can disable SCTP at a cluster level by&hellip;”

**Comment by VineethReddy02** [comment]
👋 Please rebase this PR on `dev-1.18` in order to avoid merge conflicts.

Feel free to `/hold cancel` after you've rebased.

/hold

File: content/en/docs/concepts/services-networking/service.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (janosi) [reply content/en/docs/concepts/services-networking/service.md]: Fixed.

File: content/en/docs/concepts/services-networking/service.md [hunk 3]
```diff
@@ -1144,9 +1144,9 @@ followed by the data from the client.
 
 ### SCTP
 
-{{< feature-state for_k8s_version="v1.12" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.18" state="beta" >}}
 
-Kubernetes supports SCTP as a `protocol` value in Service, Endpoint, NetworkPolicy and Pod definitions as an alpha feature. To enable this feature, the cluster administrator needs to enable the `SCTPSupport` feature gate on the apiserver, for example, `--feature-gates=SCTPSupport=true,…`.
+Kubernetes supports SCTP as a `protocol` value in Service, Endpoint, NetworkPolicy and Pod definitions as a beta feature. As a beta feature , this is enabled by default. To disable this feature, the cluster administrator needs to disable the `SCTPSupport` feature gate on the apiserver, for example, `--feature-gates=SCTPSupport=false,…`.
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md]: Is this more accurate:
```suggestion
Kubernetes has beta support for SCTP as a `protocol` value in Service, Endpoints, EndpointSlice, NetworkPolicy and Pod definitions. As a beta feature, this is enabled by default. To disable this feature, the cluster administrator needs to disable the `SCTPSupport` feature gate on the API server; for example, `--feature-gates=SCTPSupport=false,…`.
```
?

File: content/en/docs/concepts/services-networking/service.md [hunk 4]
```diff
@@ -1144,9 +1144,9 @@ followed by the data from the client.
...
 
 When the feature gate is enabled, you can set the `protocol` field of a Service, Endpoint, NetworkPolicy or Pod to `SCTP`. Kubernetes sets up the network accordingly for the SCTP associations, just like it does for TCP connections.
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md]:
```suggestion
When the feature gate is enabled, you can set the `protocol` field of a Service, Endpoints, EndpointSlice, NetworkPolicy or Pod to `SCTP`. Kubernetes sets up the network accordingly for the SCTP associations, just like it does for TCP connections.
```

File: content/en/docs/reference/command-line-tools-reference/cloud-controller-manager.md [hunk 5]
```diff
@@ -225,7 +225,7 @@ cloud-controller-manager [flags]
...
...ort=true|false (ALPHA - default=false)<br/>ServerSideApply=true|false (BETA - default=true)<br/>ServiceNodeExclusion=true|false (ALPHA - default=false)<br/>ServiceTopology=true|false (ALPHA - default=false)<br/>StartupProbe=true|false (ALPHA - default=false)<br/>StorageVersionHash=true|false (BETA - default=true)<br/>StreamingProxyRedirects=true|false (BETA - default=true)<br/>SupportNodePidsLimit=true|false (BETA - default=true)<br/>SupportPodPidsLimit=true|false (BETA - default=true)<br/>Sysctls=true|false (BETA - default=true)<br/>TTLAfterFinished=true|false (ALPHA - default=false)<br/>TaintBasedEvictions=true|false (BETA - default=true)<br/>TokenRequest=true|false (BETA - default=true)<br/>TokenRequestProjection=true|false (BETA - default=true)<br/>TopologyManager=true|false (ALPHA - default=false)<br/>ValidateProxyRedirects=true|false (BETA - default=true)<br/>VolumePVCDataSource=true|false (BETA - default=true)<br/>VolumeSnapshotDataSource=true|false (BETA - default=true)<br/>WinDSR=true|false (ALPHA - default=false)<br/>WinOverlay=true|false (ALPHA - default=false)<br/>WindowsGMSA=true|false (BETA - default=true)<br/>WindowsRunAsUserName=true|false (BETA - default=true)</td>
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/cloud-controller-manager.md]: In this PR, don't change the generated documentation; that gets updated in one batch as part of the 1.18 release.

**Comment by sftim** [comment]
@VineethReddy02 (docs release lead): watch out for:
>  this PR should not touch the generated API documentation; that change happens in a separate PR during the release itself.

As it stands this **isn't** OK to merge.

**Comment by VineethReddy02** [comment]
Hi @janosi 
If this doc is targetted for 1.18 release, Can you address the review comments asap?

**Comment by janosi** [comment]
@VineethReddy02 No, it is not targeted for 1.18 anymore.

**Comment by VineethReddy02** [comment]
As it is not targetted for release 1.18 closing this PR. As the 1.18 release is due tomorrow i.e 24th March
/close

**Pull request closed without merging**